### DISCUSSION TITLE
Loopback, and mmap?

### DISCUSSION THREAD

**Discussion by sargun** [ming1/ubdsrv#66] [General]
I'm curious why the loopback driver doesn't just mmap the whole backend in and use UBLK_IO_NEED_GET_DATA. It can update the io_desc's buffer address to the right offset in the mmap'd location, and then rely on the kernel to do the I/O.

FWIW: Isn't this effectively zero copy?

**Comment by ming1** [comment]
Yeah, in theory, this way may work, and mmap shouldn't be used, since you want to zero copy, and the kernel IO can run IO directly over the request pages from
/dev/ublkbN, just like the kernel loop dio mode.

But ublk is one generic framework, and all specific IO business needto move out of ublk_drv, so we do not want to add such specific handling for loop inside
ublk_drv(kernel side).
